.travis_update_ghpages.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+# Only publish from the main repository's master branch
+REPO_NAME="HydraCG/Specifications"
+if [ "$TRAVIS_REPO_SLUG" != "$REPO_NAME" ] || [ "$TRAVIS_BRANCH" != "master" ] || [ "$TRAVIS_PULL_REQUEST" != "false" ]; then exit; fi
+echo -e "Generating gh-pages...\n"
+
+# Checkout the gh-pages branch
+REPO_PATH=$PWD
+pushd $HOME
+git clone --quiet --branch=gh-pages https://${GH_TOKEN}@github.com/$REPO_NAME gh-pages 2>&1 > /dev/null
```

Comment:
I'm not questioning everyone's use of the `gh-pages` feature as it is an important functionality offered by GitHub. What I'm questioning is how the feature is implemented by GitHub, as I consider having a completely clean branch not related to any other branches in the same repository in any way whatsoever (source code management wise, which is what Git is for).

I think GitHub has implemented web site hosting as a hack and is also why I believe the new `docs` folder functionality has been added. Either way, this is way off topic and not something I feel is very important to agree on within this PR. Over some 🍻  sometime, perhaps? 😄